lib/AST/Module.cpp

```diff
@@ -331,8 +338,21 @@ SourceLookupCache::SourceLookupCache(const ModuleDecl &M) {
   }
 }
 
+void SourceLookupCache::addDelayedMacroExpansions() {
+  for (auto *MED : DelayedMacroExpansions) {
+    for (auto node : MED->getRewritten())
+      if (auto *decl = node.dyn_cast<Decl *>())
+        addToUnqualifiedLookupCache(TinyPtrVector<Decl *>{decl}, false);
+  }
```

Comment:
This still looks like it won't work if there are macro expansions generated by macro expansions. `addToUnqualifiedLookupCache` will append the macro declarations to `DelayedMacroExpansions`. At best, we won't add the results of rewriting the new macro expansions to the source lookup cache; at worse, `DelayedMacroExpansions` will be reallocated and we'll crash. I recommend this:

```suggestion
  for (unsigned i = 0; i != DelayedMacroExpansions.size(); ++i) {
    auto *MED = DelayedMacroExpansions[i];
    for (auto node : MED->getRewritten())
      if (auto *decl = node.dyn_cast<Decl *>())
        addToUnqualifiedLookupCache(TinyPtrVector<Decl *>{decl}, false);
  }
```